### ISSUE TITLE
Question about trimap

### ISSUE THREAD

**Issue by yarkable** [ZHKKKe/MODNet#74]
Hi, thanks for your excellent work. And I have a question, that is, in your training code, you still pass a trimap parameter to the function. So is Trimap-free algorithm still need trimap in training process but not need in inference process?

**Comment by ZHKKKe** [comment]
Hi, thanks for your attention.

We use the trimap to calculate the detail loss during training.
If you have ground truth alpha, it is easy for you to generate a trimap by using dilation and erosion, as:
```
from scipy.ndimage import grey_dilation, grey_erosion

trimap = (matte >= 0.9).astype('float32')
not_bg = (matte > 0).astype('float32')

d_size = self.im_size // 256 * random.randint(10, 20)
e_size = self.im_size // 256 * random.randint(10, 20)

trimap[np.where((grey_dilation(not_bg, size=(d_size, d_size))
    - grey_erosion(trimap, size=(e_size, e_size))) != 0)] = 0.5
```
where `matte` is the ground truth alpha normalized to `[0, 1]`, `d_size` and `e_size` are the hyper-params.

**Comment by yarkable** [comment]
So I can call this a trimap, right ? 
![matting](https://user-images.githubusercontent.com/33142987/109115273-cbddf000-7779-11eb-9013-fa182a3a6c90.png)

**Comment by ZHKKKe** [comment]
The contents in `m_d` is the same as the trimap, but the pixel values in `m_d` are different with the trimap. Following is an example of the trimap:
![matting](https://learnopencv.com/wp-content/uploads/2020/06/trimap-example.jpg)
In the image, white == 1, gray == 0.5, black == 0.

**Comment by yarkable** [comment]
I got it. Thanks for your quick reply!
